core/src/main/java/org/jboss/gwt/elemento/core/BodyObserver.java

```diff
@@ -0,0 +1,149 @@
+package org.jboss.gwt.elemento.core;
+
+import elemental2.dom.*;
+import jsinterop.base.Js;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static elemental2.dom.DomGlobal.document;
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+
+class BodyObserver {
+
+    private static String ATTACH_UID_KEY = "on-attach-uid";
+    private static String DETACH_UID_KEY = "on-detach-uid";
+
+    private static List<ElementObserver> detachObservers = new ArrayList<>();
+    private static List<ElementObserver> attachObservers = new ArrayList<>();
+
+    static {
+        MutationObserver mutationObserver = new MutationObserver((MutationRecord[] records, MutationObserver observer) -> {
+            for (int i = 0; i < records.length; i++) {
+                onElementsRemoved(records[i]);
+                onElementsAppended(records[i]);
+            }
+            return null;
+        });
+
+        MutationObserverInit mutationObserverInit = MutationObserverInit.create();
+        mutationObserverInit.setChildList(true);
+        mutationObserverInit.setSubtree(true);
+
+        mutationObserver.observe(document.body, mutationObserverInit);
+    }
+
+    private BodyObserver() {
+
+    }
+
+    private static void onElementsAppended(MutationRecord record) {
+        List<ElementObserver> observed = new ArrayList<>();
+        for (int i = 0; i < attachObservers.size(); i++) {
+            ElementObserver elementObserver = attachObservers.get(i);
+            if (isNull(elementObserver.observedElement())) {
+                observed.add(elementObserver);
+            } else {
+                if (isAppended(elementObserver.attachId())) {
+                    elementObserver.callback().onObserved(record);
+                    elementObserver.observedElement().removeAttribute(ATTACH_UID_KEY);
+                    observed.add(elementObserver);
+                }
+            }
+        }
+
+        attachObservers.removeAll(observed);
+    }
+
+    private static boolean isAppended(String attachId) {
+        return nonNull(DomGlobal.document.body.querySelector("[" + ATTACH_UID_KEY + "='" + attachId + "']"));
+    }
```

Comment:
Does it speed up things when u do not use the query selector on body and use "the same" kind of code you use for `isChildOfRemovedNode` (Iterate over the record.addedNodes and run "querySelector" for each of them)?

Could be much faster when you have a big dom!?!

Reply:
i think you have a point here, ill update this to use the same check style from detach